angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
> Found a curious edge case. If you try to pass an array of nulls ([null, null]) to this function with strictNullChecks turned on, you'll get a compilation error.

nice catch, and it definitely works fine at runtime, IFF you pass a string predicate for `expression`, but if `expression` is `x =>x.y`, it throws